pkg/api/datasources.go

```diff
@@ -515,14 +535,30 @@ func (hs *HTTPServer) UpdateDataSourceByID(c *contextmodel.ReqContext) response.
 	if err := validateJSONData(c.Req.Context(), cmd.JsonData, hs.Cfg, hs.Features); err != nil {
 		return response.Error(http.StatusBadRequest, "Failed to update datasource", err)
 	}
-
 	ds, err := hs.getRawDataSourceById(c.Req.Context(), cmd.ID, cmd.OrgID)
 	if err != nil {
 		if errors.Is(err, datasources.ErrDataSourceNotFound) {
 			return response.Error(404, "Data source not found", nil)
 		}
 		return response.Error(500, "Failed to update datasource", err)
 	}
+
+	// check if LBAC rules have been modified
+	teamHTTPheaders, err := datasources.GetTeamHTTPHeaders(ds.JsonData)
+	if err != nil {
+		return response.Error(http.StatusInternalServerError, "Failed to update datasource", err)
+	}
+	NewTeamHTTPHeaders, err := datasources.GetTeamHTTPHeaders(cmd.JsonData)
+	if err != nil {
+		return response.Error(http.StatusInternalServerError, "Failed to update datasource", err)
+	}
+	if reflect.DeepEqual(teamHTTPheaders, NewTeamHTTPHeaders) {
+		hasAccess, errAccess := evaluateTeamHTTPHeaderPermissions(hs, c, datasources.ScopePrefix+strconv.FormatInt(ds.ID, 10))
```

Comment:
`datasources.ScopePrefix` is UID based scope, we cannot use it to check permissions for the specific data source id. We have to convert data source id to UID.